tools/cldr-code/src/main/java/org/unicode/cldr/tool/CLDRModify.java

```diff
@@ -1259,6 +1236,130 @@ public void handlePath(String xpath) {
+                if (value.equals(newValue)) {
+                    return;
+                }
+                String fullXPath = cldrFileToFilter.getFullXPath(xpath);
+                replace(fullXPath, fullXPath, newValue);
+            }
+        });
+
+        // use DAIP for one thing only: replaceBaileyWithInheritanceMarker
+        fixList.add('I', "Inheritance-substitution", new CLDRFilter() {
+            private DisplayAndInputProcessor inputProcessor;
+            private final int STEPS_FROM_ROOT = 1; // only process if locale's level matches; root = 0, en = 1, ...
+
+            @Override
+            public void handleStart() {
+                int steps = stepsFromRoot(cldrFileToFilter.getLocaleID());
+                if (steps == STEPS_FROM_ROOT) {
+                    inputProcessor = new DisplayAndInputProcessor(cldrFileToFilter, true);
+                    inputProcessor.enableInheritanceReplacement(getResolved());
+                } else {
+                    inputProcessor = null;
+                }
+            }
+
+            private int stepsFromRoot(String origLoc) {
+                int steps = 0;
+                String loc = origLoc;
+                while (!LocaleNames.ROOT.equals(loc)) {
+                    loc = LocaleIDParser.getParent(loc);
+                    if (loc == null) {
+                        throw new IllegalArgumentException("Missing root in inheritance chain");
+                    }
+                    ++steps;
+                }
+                System.out.println("stepsFromRoot = " + steps + " for " + origLoc);
+                return steps;
+            }
+
+            @Override
+            public void handleEnd() {
+                inputProcessor = null; // clean up, just in case
+            }
+
+            @Override
+            public void handlePath(String xpath) {
+                if (inputProcessor == null) {
+                    return;
+                }
+                String value = cldrFileToFilter.getStringValue(xpath);
+                String newValue = inputProcessor.replaceBaileyWithInheritanceMarker(xpath, value);
+                if (value.equals(newValue)) {
+                    return;
+                }
+                String fullXPath = cldrFileToFilter.getFullXPath(xpath);
+                replace(fullXPath, fullXPath, newValue);
+            }
+        });
+
+        // Un-drop hard inheritance: revert INHERITANCE_MARKER to pre-drop-hard-inheritance values
+        fixList.add('U', "Un-drop inheritance", new CLDRFilter() {
+            // baseDir needs to be the "pre-drop" path of an existing copy of old common/main
+            // For example, 2022_10_07_pre folder gets xml from pull request 2433, commit 80029f1
+            // Also ldml.dtd is required; for example:
+            //   mkdir ../2022_10_07_pre/common/dtd
+            //   cp common/dtd/ldml.dtd ../2022_10_07_pre/common/dtd
+            final private String baseDir = "../2022_10_07_pre/";
```

Comment:
Ideally would be relative to the Generated dir, less arbitrary than `..` but not a blocker